libcodechecker/libclient/credential_manager.py

```diff
@@ -23,46 +24,54 @@
 SESSION_COOKIE_NAME = _SCN
 
 
-class UserCredentials:
+class UserCredentials(object):
 
-    def __init__(self):
+    def __init__(self, password_file_content=None, tokens=None):
         LOG.debug("Loading clientside session config.")
-
-        # Check whether user's configuration exists.
         user_home = os.path.expanduser("~")
-        session_cfg_file = os.path.join(user_home,
-                                        ".codechecker.passwords.json")
-        LOG.debug(session_cfg_file)
 
         scfg_dict = {}
-        if os.path.exists(session_cfg_file):
-            scfg_dict = load_json_or_empty(session_cfg_file, {},
-                                           "user authentication")
-        if os.path.exists(session_cfg_file):
-            check_file_owner_rw(session_cfg_file)
+
+        if password_file_content is None:
+            # Check whether user's configuration exists.
+            session_cfg_file = os.path.join(user_home,
+                                            ".codechecker.passwords.json")
+            LOG.debug(session_cfg_file)
+
+            if os.path.exists(session_cfg_file):
+                scfg_dict = load_json_or_empty(session_cfg_file, {},
+                                               "user authentication")
+            if os.path.exists(session_cfg_file):
+                check_file_owner_rw(session_cfg_file)
+
+        else:
+            scfg_dict = password_file_content
```

Comment:
This `else` can be eliminated by initalizing `scfg_dict` above with `password_file_content`:
```py
scfg_dict = password_file_content
```